src/GafferScene/Constraint.cpp

```diff
@@ -195,17 +191,21 @@ Imath::V3f interpolateConvexPolygon( const PIndexer points, const UVIndexer uvs,
 			Ae[ i ] = -( Ae[ i ] );
 		}
 
-		Av[ i ] = std::max( Av[ i ], std::numeric_limits< float >::min() );
+		Av[ i ] = std::max( Av[ i ], w * std::numeric_limits< float >::min() );
```

Comment:
Seems like this would be clearer in one assignment as `Av[ i ] = std::max( Av[ i ] / w, std::numeric_limits< float >::min() );`, rather than multiplying by w, and then immediately changing the value of `Av[ i ]` by dividing by w?

Also, might be good to have a comment about why `w` is necessary at all ... after staring at this a bit, I think I understand that if we have duplicate vertices, the loop on line 250 will skip back to the previous non-duplicated vertex for each them, meaning we repeatedly sum that edge N times for the N duplicated vertices, so it makes sense to weight down each contribution so that we end up with an average of the duplicated vertices.  This makes sense, particularly in relation to how it replaced the special case for being on a vertex, but I don't think it would be very obvious if looking at this code in the future.  Might even be good to have a different name for this variable, since `w` is used to mean something else below.